### PR TITLE
Haiku: Work around more portability issues. [skip ci]

### PR DESCRIPTION
**Pull request by infrastation**
I am going to merge these changes after re-testing one more time on Haiku hrev57585 from the next nightly build.

### CODE REVIEW

File: pcap-haiku.c [hunk 1]
```diff
@@ -288,6 +299,37 @@ pcap_activate_haiku(pcap_t *handle)
 //	#pragma mark - pcap API
 
 
+static int
+can_be_bound(const char *name)
+{
+	if (strcmp(name, "loop"))
```
Comment (guyharris) [inline pcap-haiku.c:305]: `strcmp(name, "loop") != 0` might make it a bit clearer that this is testing for names that aren't "loop".

File: pcap-haiku.c [hunk 2]
```diff
@@ -296,6 +338,11 @@ pcapint_create_interface(const char *device, char *errorBuffer)
 		         "Interface name \"%s\" is too long.", device);
 		return NULL;
 	}
+	if (! can_be_bound(device)) {
+		snprintf(errorBuffer, PCAP_ERRBUF_SIZE,
+		         "Interface \"%s\" is not supported.", device);
```
Comment (guyharris) [inline pcap-haiku.c:343]: Perhaps "does not support capturing traffic" - it's "supported" in that you san presumably send packets to 127.0.0.1, but you can't see those packets in a capture in older versions of Haiku.

**Comment by infrastation** [comment]
Updated as suggested.

**Comment by guyharris** [comment]
> Still waiting on the development snapshot.

Presumably "the development snapshot" means Haiku hrev57585 from the next nightly build.

**Comment by infrastation** [comment]
hrev57585 or later (the nightly builds stopped at hrev57581, see haiku/infrastructure#114). I am trying to build hrev57588 locally.

**Comment by guyharris** [comment]
Should this be backported to 1.10?

**Comment by guyharris** [comment]
...along with the previous improvements?

**Comment by infrastation** [comment]
Yes, the recent change set makes tcpdump more useful. Let's give at least a few days for the dust to settle. Besides, I considered to validate packet AF for the DLT, and to size the buffer based on the MTU, but both directions were not fruitful for reasons I would like to document in the source code comments.

**Comment by infrastation** [comment]
Indeed there is a minor bug in one of my recent commits, I am going to include the bug fix into these fixups.

**Comment by infrastation** [comment]
Committed and now looks ready for back-porting to me. If anybody else can proof-read the recent Haiku commits, please do.

**Comment by infrastation** [comment]
The change set has been backported to the 1.10 branch, libpcap compiles and works as expected on both R1/beta4 and the current master branch (hrev57592).

**Pull request merged**